compiler/src/dotty/tools/dotc/ast/Desugar.scala

```diff
@@ -232,34 +226,82 @@ object desugar {
   private def defDef(meth: DefDef, isPrimaryConstructor: Boolean = false)(using Context): Tree =
     addDefaultGetters(elimContextBounds(meth, isPrimaryConstructor))
 
-  private def elimContextBounds(meth: DefDef, isPrimaryConstructor: Boolean)(using Context): DefDef =
-    val DefDef(_, paramss, tpt, rhs) = meth
-    val evidenceParamBuf = ListBuffer[ValDef]()
+  /** Drop context bounds in given TypeDef, replacing them with evidence ValDefs that
+   *  get added to a buffer.
+   *  @param tdef          The given TypeDef
+   *  @param evidenceBuf   The buffer to which evidence gets added. This buffer
+   *                       is shared between desugarings of different type parameters
+   *                       of the same method.
+   *  @param evidenceFlags The flags to use for evidence definitions
+   *  @param freshName     A function to generate fresh names for evidence definitions
+   *  @param allParams     If `tdef` is a type paramter, all parameters of the owning method,
+   *                       otherwise the empty list.
+   */
+  private def desugarContextBounds(
+      tdef: TypeDef,
+      evidenceBuf: ListBuffer[ValDef],
+      evidenceFlags: FlagSet,
+      freshName: untpd.Tree => TermName,
+      allParamss: List[ParamClause])(using Context): TypeDef =
 
-    var seenContextBounds: Int = 0
-    def desugarContextBounds(rhs: Tree): Tree = rhs match
+    val evidenceNames = ListBuffer[TermName]()
+
+    def desugarRhs(rhs: Tree): Tree = rhs match
```

Comment:
```suggestion
    def desugarRHS(rhs: Tree): Tree = rhs match
```
For consistency with other occurrences of abbreviations in camel-cased names.